### PR TITLE
Layout

### PR DESCRIPTION
**Pull request by fifikim**
This PR corresponds to the [ticket for Create app layout with mock data ](https://trello.com/c/irPlkxvq)

After discussing with Brian we modified the scope of this ticket to include:
- `navbar`, `dashboard`, and `browse` components

The search bar and results will be located on a separate page & implemented in the next ticket along with routing.

Updated preview:
image

### CODE REVIEW

**Review by seoulection — APPROVED** [review]
Looks good to me. Just one question though - is the dashboard ultimately going to the be metrics for repos that a user searches for? Or simply put, what is the dashboard?

**Comment by fifikim** [comment]
> Looks good to me. Just one question though - is the dashboard ultimately going to the be metrics for repos that a user searches for? Or simply put, what is the dashboard?

I think we decided that the dashboard would be the section that displays cumulative stats about the entire 8L account. The mock up of that section could prob be a lot more descriptive but I was planning on asking you all what kind of info you wanted to see there when I get to that part

File: src/app/repo-preview/repo-preview.component.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div *ngIf="repo">
+  <a href="/">
+    <div class="card">
+      <h3 class="repo-name">{{repo.name}}</h3>
+      <p class="repo-description">Description: {{repo.description}}</p>
```
Comment (ksdenisova) [inline src/app/repo-preview/repo-preview.component.html]: What is the expected behavior when there is no description in the repository?

File: src/app/repo-preview/repo-preview.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    component.repo = REPOSITORIES[0];
+    fixture.detectChanges();
+  });
+
+  it('should create the RepoPreview component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should receive a repo as input and display its name', () => {
+    expect(fixture.nativeElement.querySelector('.repo-name').textContent).toContain('zagaku');
```
Comment (ksdenisova) [inline src/app/repo-preview/repo-preview.component.spec.ts]: Could you extract expectations as variables where possible?

File: src/app/repository.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export interface Repository {
+  name: string;
+  description: string | null;
```
Comment (ksdenisova) [inline src/app/repository.ts]: Have you considered making some fields optional instead of having an empty value?

File: src/app/repository.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  collaborators: number | null;
+  openPRs: number;
+  lastCommit: Commit | null;
+}
+
+export interface Commit {
+  message: string;
+  author: string;
+  date: string;
+}
```
Comment (ksdenisova) [inline src/app/repository.ts]: A linter rule for checking an empty line at the end of the file could be helpful.

**Review by ksdenisova — APPROVED** [review]
I left a few non-blocking comments, but PR looks good to me

File: src/app/repo-preview/repo-preview.component.html (hunk 1 above)
Comment (fifikim) [reply src/app/repo-preview/repo-preview.component.html]: Ah, I added a mapper to handle this in the next ticket but for now I'll update this to print "not available" if there is no description.

File: src/app/repo-preview/repo-preview.component.spec.ts (hunk 2 above)
Comment (fifikim) [reply src/app/repo-preview/repo-preview.component.spec.ts]: yes! done

File: src/app/repository.ts (hunk 4 above)
Comment (fifikim) [reply src/app/repository.ts]: added!

File: src/app/repository.ts (hunk 3 above)
Comment (fifikim) [reply src/app/repository.ts]: I had not considered that but I like it! updated

**Pull request merged**